### PR TITLE
Empty Struct to Namespace

### PR DESCRIPTION
**Pull request by Jim-Marsden**
* [x] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before? 
Issue #2224
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

I will be empty structs to namespaces.

This PR is related to the issue #

## Tasks

* [ ] Tested on Linux
* [x] Tested on Windows
* [x] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

Describe how to best test these changes. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve) if possible, you can use the follow template as a start:

```cpp
// pending
```

### CODE REVIEW

File: include/SFML/Window/Joystick.hpp [hunk 1]
```diff
@@ -88,7 +85,7 @@ class SFML_WINDOW_API Joystick
     /// \return True if the joystick is connected, false otherwise
     ///
     ////////////////////////////////////////////////////////////
-    static bool isConnected(unsigned int joystick);
+    bool isConnected(unsigned int joystick);
```
Comment (ChrisThrasher) [inline include/SFML/Window/Joystick.hpp:88]: I think you'll need to mark all these functions with `SFML_WINDOW_API` to fix shared library linking issues.

Comment (Jim-Marsden) [reply include/SFML/Window/Joystick.hpp:88]: That would make sense. I was trying to figure out what I needed to do. Thank you!

**Comment by Jim-Marsden** [comment]
To clarify unless you would prefer to keep all of these commits, I'll squash them before I turn this into a proper PR.

File: test/install/Install.cpp [hunk 2]
```diff
@@ -7,9 +7,10 @@ int main()
 {
     // Audio
     [[maybe_unused]] sf::InputSoundFile inputSoundFile;
-    [[maybe_unused]] sf::Listener       listener;
-    [[maybe_unused]] sf::Music          music;
-    [[maybe_unused]] sf::Sound          sound;
+    // TODO: Should I change this to another test like "auto volume{sf::Listen::getGlobalVolume()};" or remove it?
+    // [[maybe_unused]] sf::Listener       listener;
```
Comment (ChrisThrasher) [inline test/install/Install.cpp:11]: Funny how I accidentally used one of these pseudo-classes. Just replace `listener` with an instance of some other Audio module type.

Comment (Jim-Marsden) [reply test/install/Install.cpp]: I feel validated with this PR as a result.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by kimci86 — CHANGES_REQUESTED** [review]
Doxygen comment blocks need to be updated: `\class` becomes `\namespace`.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
Screenshot 2022-12-28 at 5 33 43 PM

The generated docs look okay. `sf::Joystick` appears as a namespace under the "Classes" tab. Not sure why it doesn't appear under the list of namespaces though.

**Comment by Jim-Marsden** [comment]
I changed a bit of documentation, please be harsh.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]
Thanks for finding and fixing all these ~~classes~~ namespaces!

I took my time to make sure we didn't miss any issues with the docs but as far as I can tell, the docs have lost no information and now correctly refer to these namespaces as namespaces instead of classes.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]
Any reason `System/SuspendAwareClock.hpp` wasn't included in this PR?

**Comment by ChrisThrasher** [comment]
> Any reason `System/SuspendAwareClock.hpp` wasn't included in this PR?

I believe this has to remain a class.

**Comment by Jim-Marsden** [comment]
> Any reason `System/SuspendAwareClock.hpp` wasn't included in this PR?

It is not in the same group of "This is a namespace masquerading as a struct". I thought it was at first.

**Comment by ChrisThrasher** [comment]
Thank you!!

**Pull request merged**